### PR TITLE
[SPARK-12461] [SQL] Add ExpressionDescription to math functions

### PR DESCRIPTION
**Pull request by vectorijk**
cc @yhuai

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala [hunk 1]
```diff
@@ -218,10 +257,22 @@ case class Conv(numExpr: Expression, fromBaseExpr: Expression, toBaseExpr: Expre
   }
 }
 
+@ExpressionDescription(
+  usage = "_FUNC_(value) - Returns Euler's number e raised to the power of a double value.",
+  extended = "> SELECT _FUNC_(1.0);\n2.718281828459045")
 case class Exp(child: Expression) extends UnaryMathExpression(math.exp, "EXP")
 
+@ExpressionDescription(
+  usage = "_FUNC_(value) - Returns e^x -1.",
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala]: add a space after -

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala [hunk 2]
```diff
@@ -279,6 +330,9 @@ object Factorial {
   )
 }
 
+@ExpressionDescription(
+  usage = "_FUNC_(value) - Returns Factorial of the number",
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala]: specify the valid range here?

Also might be more useful to say

Returns the n factorial. n is valid in [0, 20] range.

Comment (vectorijk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala]: Thanks @rxin! I have updated.

**Comment by vectorijk** [comment]
I will close this PR at this time. If needed, I might re-open this.

**Pull request closed without merging**